app/src/main/java/com/duckduckgo/app/bookmarks/ui/bookmarkfolders/BookmarkFoldersViewModel.kt

```diff
@@ -35,16 +37,21 @@ import javax.inject.Provider
 class BookmarkFoldersViewModel(
     val bookmarksRepository: BookmarksRepository,
     private val dispatcherProvider: DispatcherProvider
-) : ViewModel() {
+) : ViewModel(), AddBookmarkFolderListener {
 
     data class ViewState(
-        val folderStructure: List<BookmarkFolderItem> = emptyList()
+        val folderStructure: List<BookmarkFolderItem> = emptyList(),
+        var currentFolder: BookmarkFolder? = null,
```

Comment:
Sorry, can you explain a bit why we have added these 2 new properties to the viewstate? what are we trying to achieve?

Reply:
In the process of understanding and making things work (was not able to understand the schema completely), primarily making use of the `getFlatFolderStructure` function these two properties were required.

But they are not required in the ViewState. I'll add them as local variables inside the viewModel.

Reply:
I'm still not sure about the correctness of the implementation.

The major change in this class is the usage of `AddBookmarkFolderListener` to listen to the addition of a new bookmark folder and in turn update the view i.e the "flat folder structure".

As I was not able to understand what's happening in `getFlatFolderStructure` inside the `BookmarksDataRepository` I was only able to reuse the function.

To reuse the function there is a `rootFolderName` property required. Which was initially obtained from the string value (maybe to make translations compatible). So I did the same using a new Command object `NewFolderCreatedUpdateTheStructure`.

Reply:
@4shutosh I've been taking a look at this part. And I understand better what was your problem here.

This is my suggestion to simplify this:
- keep the `viewstate` as it was before. No need to add new properties.
- change the signature of `newFolderAdded` to receive those params from the activity.

Doing that you will simplify a lot this class.